---
title: "Step 1: Sample QC"
author: "Belinda Cornes"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

#setwd("/projects/compsci/corneb_cs/workflowr/Thaiss/Thaiss_workflowr_2/")

#libraries
library(broman)
library(qtl2)
library(qtlcharts)
library(ggplot2)
library(ggrepel)
#library(DOQTL)
library(mclust)
#source("code/reconst_utils.R")
library("kableExtra")
library("knitr")
library("fst")
library(data.table) 
library(tidyr)     
library(mclust)     
library(rhdf5)      
library(optparse)
library(dplyr)
library(cluster)

```
This script is running genotype QC on raw data (with some outcomes already seen in the [project at a glace](summary.html)). Here, we first load the R/qtl2 package and the data.
Weâ€™ll also load the R/broman package for some utilities and plotting functions, and R/qtlcharts for interactive graphs.

#### Quick Facts:
+ no samples have been removed yet
+ total number of samples == 112
+ total number of samples with phenotypes == 106 (will not be reflected below)
+ total number with both geno & pheno == 106
+ three (3) phenotypes: `Time_min` (transformed), `Distance_m`, `Energy_J`
+ three (3) covariates (`sex` & `DO generation` as well as an extra sex assignment, `sexp`, from sample phenotypic meta-data)


We will follow the steps by `Karl Broman` found [here](https://kbroman.org/qtl2/assets/vignettes/do_diagnostics.html)

### Loading Project
```{r generate json file for all batches, eval=TRUE}

gm <- get(load("data/gm_DO112.RData"))

gm
````

### Missing Data
```{r Missing data per sample, fig.height = 6, fig.width = 9.5, fig.align = "center"}

percent_missing <- n_missing(gm, "ind", "prop")*100

labels <- paste0(names(percent_missing), " (", round(percent_missing,2), "%)")
iplot(seq_along(percent_missing), percent_missing, indID=labels,
      chartOpts=list(xlab="Mouse", ylab="Percent missing genotype data",
                     ylim=c(0, 60)))

#save into pdf
pdf(file = "output/Percent_missing_genotype_data.pdf", width = 20, height = 20)
labels <- as.character(do.call(rbind.data.frame, strsplit(ind_ids(gm), "V01_"))[,2])
labels[percent_missing < 5] = ""
# Change point shapes and colors
p <- ggplot(data = data.frame(Mouse=seq_along(percent_missing),  
                         Percent_missing_genotype_data = percent_missing,
                         batch = factor(as.character(do.call(rbind.data.frame, strsplit(ind_ids(gm), "_"))[,6]))), 
        aes(x=Mouse, y=Percent_missing_genotype_data, color = batch)) +
  geom_point() +
  geom_hline(yintercept=5, linetype="solid", color = "red") +
  geom_text_repel(aes(label=labels), vjust = 0, nudge_y = 0.01, show.legend = FALSE, size=3) +
  theme(text = element_text(size = 10))
p

dev.off()

p

save(percent_missing,
     file = "data/percent_missing_id.RData")

gm.covar = data.frame(id=rownames(gm$covar),gm$covar)
qc_info_cr <- merge(gm.covar,
	                data.frame(id = names(percent_missing),percent_missing = percent_missing,stringsAsFactors = F),by = "id")
bad.sample.cr <- qc_info_cr[qc_info_cr$percent_missing >= 10,]

```

There are 3 mice that are missing >5% of their genotypes. These samples are:

```{r high missing table, echo=FALSE}

rownames(bad.sample.cr) <- NULL
bad.sample.cr[] <- lapply(bad.sample.cr, as.character)
bad.sample.cr$Neogen_Sample_ID <- sapply(strsplit(as.character(bad.sample.cr$id),'_',fixed=T),function(x) x[7])

bad.sample.cr[c("Neogen_Sample_ID","percent_missing")] %>%
  kable(escape = F,align = c("cc"),linesep ="\\hline") %>%
  kable_styling(full_width = F) %>%
  kable_styling("striped", full_width = F) 

```

```{r removing bad samples CR, include=FALSE}

##removing bad samples
#gm <- gm[paste0("-",as.character(bad.sample.cr$id)),]#

#gm

```

### Sex
```{r Sexes, fig.height = 6, fig.width = 9.5, fig.align = "center"}

xint <- read_csv_numer("data/Univ_of_Penn_Thaiss_qtl2_chrXint.csv", transpose=TRUE)
yint <- read_csv_numer("data/Univ_of_Penn_Thaiss_qtl2_chrYint.csv", transpose=TRUE)

#sex order
sex <- gm$covar[rownames(xint),"sexp"]
#sex <- substr(rownames(xint), 1, 1)

x_pval <- apply(xint, 2, function(a) t.test(a ~ sex)$p.value)
y_pval <- apply(yint, 2, function(a) t.test(a ~ sex)$p.value)

xint_ave <- rowMeans(xint[, x_pval < 0.05/length(x_pval)], na.rm=TRUE)
yint_ave <- rowMeans(yint[, y_pval < 0.05/length(y_pval)], na.rm=TRUE)

#BAF = yint / (xint + yint)
#LRR = log2((xint + yint) / (xint_ave+yint_ave))

point_colors <- as.character( brocolors("web")[c("green", "purple","red")] )
percent_missing <- n_missing(gm, summary="proportion")*100
labels <- paste0(names(xint_ave), " (", round(percent_missing), "%)")
iplot(xint_ave, yint_ave, group=sex, indID=labels,
      chartOpts=list(pointcolor=point_colors, pointsize=4,
                     xlab="Average X chr intensity", ylab="Average Y chr intensity"))
```

For figures above and below, those labelled as female in metadata given, are coloured `green`, with those labelled as male are coloured as `purple`. The above is an interactive scatterplot of the average SNP intensity on the Y chromosome versus the average SNP intensity on the X chromosome.

The cluster in the upper-left are male mice (low X chromosome intensity and high Y chromosome intensity), and the cluster in the lower-right are female mice (high X chromosome intensity and low Y chromsome intensity).  

```{r Sexes p2, fig.height = 6, fig.width = 9.5, fig.align = "center"}

phetX <- rowSums(gm$geno$X == 2)/rowSums(gm$geno$X != 0)
phetX <- phetX[names(phetX) %in% names(xint_ave)]
iplot(xint_ave, phetX, group=sex, indID=labels,
      chartOpts=list(pointcolor=point_colors, pointsize=4,
                     xlab="Average X chr intensity", ylab="Proportion het on X chr"))
```

In the above scatterplot, we show the proportion of hets vs the average intensity for the X chromosome SNPs. In calculating the proportion of heterozygous genotypes for the individuals, we look at X chromosome genotypes equal to 2 which corresponds to the heterozygote) relative to not being 0 (which is used to encode missing genotypes). The genotypes are arranged with rows being individuals and columns being markers. 

The following are the mice that have had sex incorrectly assigned:

```{r sex diff table, echo=FALSE}

bad.sample.sex <- subset(gm$covar, (gm$covar$sex != gm$covar$sexp) | is.na(gm$covar$sexp))
bad.sample.sex$sexp[is.na(bad.sample.sex$sexp)] <- '--'
bad.sample.sex$Neogen_Sample_ID <- sapply(strsplit(rownames(bad.sample.sex),'_',fixed=T),function(x) x[7])
rownames(bad.sample.sex) <- NULL

bad.sample.sex[c("Neogen_Sample_ID","sex","sexp")] %>%
  kable(escape = F,align = c("ccc"),linesep ="\\hline") %>%
  kable_styling(full_width = F) %>%
  kable_styling("striped", full_width = F) 

```

### Sample Duplicates
```{r Sample duplicates}

cg <- compare_geno(gm, cores=10)
summary.cg <- summary(cg)
```
```{r Sample duplicates table, echo=FALSE}
rownames(summary.cg) <- NULL
summary.cg$iid1 <- sapply(strsplit(as.character(summary.cg$ind1),'_',fixed=T),function(x) x[7]) 
summary.cg$iid2 <- sapply(strsplit(as.character(summary.cg$ind2),'_',fixed=T),function(x) x[7]) 

summary.cg[c(9,10,3:6)] %>%
  kable(escape = F,align = c("llcccccc"),linesep ="\\hline") %>%
  kable_styling(full_width = F) %>%
  kable_styling("striped", full_width = F) 

```

Here is a histogram of the proportion of matching genotypes. The tick marks below the histogram indicate individual pairs.

```{r Sample duplicates figures}
save(summary.cg,
     file = "data/summary.cg.RData")

pdf(file = "output/Proportion_matching_genotypes_before_removal_of_bad_samples.pdf", width = 20, height = 20) 
par(mar=c(5.1,0.6,0.6, 0.6))
hist(cg[upper.tri(cg)], breaks=seq(0, 1, length=201),
     main="", yaxt="n", ylab="", xlab="Proportion matching genotypes")
rug(cg[upper.tri(cg)])
dev.off()

par(mar=c(5.1,0.6,0.6, 0.6))
hist(cg[upper.tri(cg)], breaks=seq(0, 1, length=201),
     main="", yaxt="n", ylab="", xlab="Proportion matching genotypes")
rug(cg[upper.tri(cg)])

```

The majority of pairs share about 50% genotypes, but there is a group of pairs that are more closely related and share about 70% genotypes. The pairs cited above have >90% matching genotypes.  

```{r Sample duplicates 2 figures, include=FALSE}

#cgsub <- cg[percent_missing < 50, percent_missing < 50]
#par(mar=c(5.1,0.6,0.6, 0.6))
#hist(cgsub[upper.tri(cgsub)], breaks=seq(0, 1, length=201),
#     main="", yaxt="n", ylab="", xlab="Proportion matching genotypes")
#rug(cgsub[upper.tri(cgsub)])

```

```{r top 20 high missing rate, include = FALSE}
##show top 20 samples with missing genotypes
#percent_missing <- n_missing(gm, "ind", "prop")*100
#round(sort(percent_missing, decreasing=TRUE)[1:20], 1)  
```

```{r removing duplicate samples, include=FALSE}
#removing duplcate pairs

summary.cg$Name.ind1 <- as.character(do.call(rbind.data.frame, strsplit(as.character(summary.cg$ind1), "_"))[,6])
summary.cg$Name.ind2 <- as.character(do.call(rbind.data.frame, strsplit(as.character(summary.cg$ind2), "_"))[,6])
summary.cg$miss.ind1 <- percent_missing[match(summary.cg$ind1, names(percent_missing))]
summary.cg$miss.ind2 <- percent_missing[match(summary.cg$ind2, names(percent_missing))]
summary.cg$phen.ind1 <- gm$pheno[match(summary.cg$ind1, rownames(gm$pheno))]
summary.cg$phen.ind2 <- gm$pheno[match(summary.cg$ind2, rownames(gm$pheno))]
summary.cg$remove.id <- ifelse((summary.cg$miss.ind1 > summary.cg$miss.ind2), summary.cg$ind1, summary.cg$ind2)
#summary.cg$remove.id.p1 <- ifelse(is.na(summary.cg$phen.ind1) & (is.na(summary.cg$phen.ind1) | !is.na(summary.cg$phen.ind1)), summary.cg$ind1, '')
#summary.cg$remove.id.p2 <- ifelse(is.na(summary.cg$phen.ind2) & (is.na(summary.cg$phen.ind1) | !is.na(summary.cg$phen.ind1)), summary.cg$ind2, '')
#remove.IDs <- rbind(summary.cg$remove.id.m[-1],summary.cg$remove.id.p1[-1],summary.cg$remove.id.p2[-1])
summary.cg$remove.id  

#qc_info$remove.id.duplicated <- ifelse(qc_info$id %in% summary.cg$remove.id, TRUE,FALSE)

#bad.sample.cr.dup <- NULL

#bad.sample.cr.dup$id <- c("Univ_of_Penn_Thaiss_MURMUGV01_20200130_9_A6",#
#	                     "Univ_of_Penn_Thiass_MURGIGV01_20200226_90_C12", 
#	                     "Univ_of_Penn_Thaiss_MURMUGV01_20200130_109_G2",
#	                     "Univ_of_Penn_Thaiss_MURMUGV01_20200130_111_B3",
#	                     "Univ_of_Penn_Thaiss_MURMUGV01_20200130_110_H5",
#	                     "Univ_of_Penn_Thaiss_MURMUGV01_20200130_112_F5")

#gm <- gm[paste0("-",as.character(bad.sample.cr.dup$id)),]

#gm

#percent_missing <- n_missing(gm, "ind", "prop")*100
#round(sort(percent_missing, decreasing=TRUE)[1:19], 1)

```

### Array Intensities
```{r Array intensities, fig.height = 6, fig.width = 9.5, fig.align = "center"}

#load the intensities.fst.RData
load("data/intensities.fst.RData")
#X and Y channel
X <- result[result$channel == "X",]
rownames(X) <- X$snp
X <- X[,c(-1,-2)]

Y <- result[result$channel == "Y",]
rownames(Y) <- Y$snp
Y <- Y[,c(-1,-2)]

int <- result

rm(result)
int <- int[seq(1, nrow(int), by=2),-(1:2)] + int[-seq(1, nrow(int), by=2),-(1:2)]
int <- int[,intersect(ind_ids(gm), colnames(int))]
n <- names(sort(percent_missing[intersect(ind_ids(gm), colnames(int))], decreasing=TRUE))
iboxplot(log10(t(int[,n])+1), orderByMedian=FALSE, chartOpts=list(ylab="log10(SNP intensity + 1)"))
```
In the above plot, distributions of array intensities (after a log10(x+1) transformation) are displayed. 

The arrays are sorted by the proportion of missing genotype data for the sample, and the curves connect various quantiles of the intensities.

```{r Array intensities percentilw figure, fig.height = 6, fig.width = 9.5, fig.align = "center"}

qu <- apply(int, 2, quantile, c(0.01, 0.99), na.rm=TRUE)
group <- (percent_missing >= 19.97) + (percent_missing > 5) + (percent_missing > 2) + 1
labels <- paste0(colnames(qu), " (", round(percent_missing), "%)")
iplot(qu[1,], qu[2,], indID=labels, group=group,
      chartOpts=list(xlab="1 %ile of array intensities",
                     ylab="99 %ile of array intensities",
                     pointcolor=c("#ccc", "slateblue", "Orchid", "#ff851b")))
```
For this particular set of arrays, a plot of the 1 %ile vs the 99 %ile is quite revealing. In the following, the orange points are those with > 20% missing genotypes, the pink points are the samples with 5-20% missing genotypes, and the blue points are the samples with 2-5% missing genotypes.

### Genotype Frequencies

The following triangle plots show the genotype frequency distributions for the mice, among the four groups of markers with common minor allele frequency (MAF) in the founder strains. These plots make use of the fact that for a point within an equilateral triangle, the sum of the distances to the three sides is a constant.

```{r Genotype frequencies}
g <- do.call("cbind", gm$geno[1:19])
fg <- do.call("cbind", gm$founder_geno[1:19])
g <- g[,colSums(fg==0)==0]
fg <- fg[,colSums(fg==0)==0]
fgn <- colSums(fg==3)

gf_ind <- vector("list", 4)
for(i in 1:4) {
  gf_ind[[i]] <- t(apply(g[,fgn==i], 1, function(a) table(factor(a, 1:3))/sum(a != 0)))
}

par(mfrow=c(4,1), mar=c(0.6, 0.6, 2.6, 0.6))
for(i in 1:4) {
  triplot(c("AA", "AB", "BB"), main=paste0("MAF = ", i, "/8"))
  tripoints(gf_ind[[i]], pch=21, bg="lightblue")
  tripoints(c((1-i/8)^2, 2*i/8*(1-i/8), (i/8)^2), pch=21, bg="violetred")
  
  if(i>=3) { # label mouse with lowest het
    wh <- which(gf_ind[[i]][,2] == min(gf_ind[[i]][,2]))
    tritext(gf_ind[[i]][wh,,drop=FALSE] + c(0.02, -0.02, 0),
            names(wh), adj=c(0, 1))
  }
  
  # label other mice
  if(i==1) {
    lab <- rownames(gf_ind[[i]])[gf_ind[[i]][,2]>0.3]
  }
  else if(i==2) {
    lab <- rownames(gf_ind[[i]])[gf_ind[[i]][,2]>0.48]
  }
  else if(i==3) {
    lab <- rownames(gf_ind[[i]])[gf_ind[[i]][,2]>0.51]
  }
  else if(i==4) {
    lab <- rownames(gf_ind[[i]])[gf_ind[[i]][,2]>0.6]
  }
  
  for(ind in lab) {
    if(grepl("^F", ind) && i != 3) {
      tritext(gf_ind[[i]][ind,,drop=FALSE] + c(-0.01, 0, +0.01), ind, adj=c(1,0.5))
    } else {
      tritext(gf_ind[[i]][ind,,drop=FALSE] + c(0.01, 0, -0.01), ind, adj=c(0,0.5))
    }
  }
}

````

The majority of individuals are tightly clustered around the expected distribution (in pink). There are a few outliers with elevated & reduced heterozygosity, but most of these are among the samples with highest rates of missing genotypes. 

```{r removing low af, include=FALSE}

#bad.sample.cr.dup.gf <- NULL

#bad.sample.cr.dup.gf$id <- c("Univ_of_Penn_Thaiss_MURMUGV01_20200130_108_D12")

#gm <- gm[paste0("-",as.character(bad.sample.cr.dup.gf$id)),]

#gm

```

### Crossover Counts
```{r Crossover counts, fig.height = 6, fig.width = 9.5, fig.align = "center"}
#load pre-caluated results
load("data/pr.RData")
load("data/m.RData")
load("data/nxo.RData")

#crossover
totxo <- rowSums(nxo)[rownames(gm$covar)]
iplot(seq_along(totxo),
      totxo,
      group=gm$covar$ngen,
      chartOpts=list(xlab="Mouse", ylab="Number of crossovers", 
                     margin=list(left=80,top=40,right=40,bottom=40,inner=5),
                     axispos=list(xtitle=25,ytitle=50,xlabel=5,ylabel=5)))

#save crossover into pdf
pdf(file = "output/number_crossover.pdf")
cross_over <- data.frame(Mouse = seq_along(totxo), Number_crossovers = totxo, generation = gm$covar$ngen)
names(totxo) <- as.character(do.call(rbind.data.frame, strsplit(names(totxo), "V01_"))[,2])
names(totxo)[totxo >= 200 & totxo <= 800] = ""
# Change point shapes and colors
p <-ggplot(cross_over, aes(x=Mouse, y=Number_crossovers, fill = generation, color=generation)) +
  geom_point() +
  geom_text_repel(aes(label=names(totxo),hjust=0,vjust=0), show.legend = FALSE)
p
dev.off()

p

```
Counts of inferred crossovers can be a useful diagnostic for identifying problem samples, which may show an excessive number of apparent crossovers. Above is a plot of the number of crossovers vs the mouse index, colored by generation.

Weâ€™re looking for mice with an excessive number of crossovers as well as mice with very few crossovers.  These are shown below

**NB:** DO generation of an individual is taken into account. 

```{r  corssover counts, echo = FALSE}
#Here are the crossover counts for those  mice:
tmp <- cbind(percent_missing=round(percent_missing[rownames(gm$covar)],2), total_xo=totxo)[percent_missing[rownames(gm$covar)] >= 5,]
tmp[order(tmp[,1]),]

##number of crossovers

totxo <- rowSums(nxo)[rownames(gm$covar)]
totxot <- totxo[totxo <= 200 | totxo >= 1000]

totxot <- as.data.frame(totxot)
#totxot <- as.data.frame(t(totxot))

totxot$Neogen_Sample_ID <- sapply(strsplit(rownames(totxot),'_',fixed=T),function(x) x[7]) 
totxot$number.xo <- totxot$totxot

rownames(totxot) <- NULL

totxot[c(2,3)] %>%
  kable(escape = F,align = c("cc"),linesep ="\\hline") %>%
  kable_styling(full_width = F) %>%
  kable_styling("striped", full_width = F) 

```

### Genotyping Error LOD Scores
```{r  Genotyping error LOD scores, fig.height = 6, fig.width = 9.5, fig.align = "center"}
load("data/e.RData")
errors_ind <- rowSums(e>2)[rownames(gm$covar)]/n_typed(gm)*100
lab <- paste0(names(errors_ind), " (", myround(percent_missing[rownames(gm$covar)],1), "%)")
iplot(seq_along(errors_ind), errors_ind, indID=lab,
      chartOpts=list(xlab="Mouse", ylab="Percent genotyping errors", ylim=c(0, 8),
                     axispos=list(xtitle=25, ytitle=50, xlabel=5, ylabel=5)))
save(errors_ind, file = "data/errors_ind.RData")

```


### Apparent Genotyping Errors
```{r apparent genotyping errors, fig.height = 6, fig.width = 9.5, fig.align = "center"}

load("data/snpg.RData")

snpgn <- snpg[rownames(gm$covar),]

gobs <- do.call("cbind", gm$geno)
gobs[gobs==0] <- NA

par(pty="s")
err_direct <- rowMeans(snpgn != gobs, na.rm=TRUE)*100
errors_ind_0 <- rowSums(e > 0)[rownames(gm$covar)]/n_typed(gm)*100
par(mar=c(4.1,4.1,0.6, 0.6))
grayplot(errors_ind_0, err_direct,
         xlab="Percent errors (error LOD > 0)",
         ylab="Percent errors (obs vs predicted)",
         xlim=c(0, 2), ylim=c(0, 2))
abline(0,1,lty=2, col="gray60")

pdf(file = "output/Percent_genotype_errors_obs_vs_predicted.pdf",width = 20, height = 20) 
par(pty="s")
err_direct <- rowMeans(snpgn != gobs, na.rm=TRUE)*100
errors_ind_0 <- rowSums(e > 0)[rownames(gm$covar)]/n_typed(gm)*100
par(mar=c(4.1,4.1,0.6, 0.6))
grayplot(errors_ind_0, err_direct,
         xlab="Percent errors (error LOD > 0)",
         ylab="Percent errors (obs vs predicted)",
         xlim=c(0, 2), ylim=c(0, 2))
abline(0,1,lty=2, col="gray60")
dev.off()

par(pty="s")
par(mar=c(4.1,4.1,0.6, 0.6))
grayplot(errors_ind_0, err_direct,
         xlab="Percent errors (error LOD > 0)",
         ylab="Percent errors (obs vs predicted)",
         xlim=c(0, 0.4), ylim=c(0, 0.4))
abline(0,1,lty=2, col="gray60")

```

There were no samples with genotype errors greater than 1%

### Removing Samples
```{r removing samples }

#percent missing
gm.covar = data.frame(id=rownames(gm$covar),gm$covar)
qc_info <- merge(gm.covar,
	                data.frame(id = names(percent_missing),percent_missing = percent_missing,stringsAsFactors = F),by = "id")

#cross_over
qc_info <- merge(qc_info,
                 data.frame(id = rownames(cross_over),
                            Number_crossovers = cross_over$Number_crossovers,stringsAsFactors = F),by = "id")

#missing sex
qc_info$sex.match <- ifelse(qc_info$sexp == qc_info$sex, TRUE, FALSE)

#genotype errors
qc_info <- merge(qc_info,
                 data.frame(id = names(errors_ind),
                            genotype_erros = errors_ind,stringsAsFactors = F),by = "id")

#duplicated id to be remove
qc_info$remove.id.duplicated <- ifelse(qc_info$id %in% summary.cg$remove.id, TRUE,FALSE)

bad.sample <- qc_info[qc_info$ngen ==1 | qc_info$Number_crossovers <= 200 | qc_info$Number_crossovers >=1000 | qc_info$percent_missing >= 10 | qc_info$genotype_erros >= 1 | qc_info$remove.id.duplicated == TRUE,]

save(qc_info, bad.sample, file = "data/qc_info_bad_sample.RData")

gm_DO112_samqc <- gm[paste0("-",as.character(bad.sample$id)),]

#fix sex mismatch
gm_DO112_samqc$covar[gm_DO112_samqc$covar$id %in% names(yint_ave[!names(yint_ave) %in% bad.sample$id][yint_ave[!names(yint_ave) %in% bad.sample$id] <= 0.1]),"sexp"] <- "F"

gm_DO112_samqc

save(gm_DO112_samqc, file = "data/gm_DO112_samqc.RData")

# update other stuff
e <- e[ind_ids(gm_DO112_samqc),]
g <- g[ind_ids(gm_DO112_samqc),]
snpg <- snpg[ind_ids(gm_DO112_samqc),]

save(e,g,snpg, file = "data/e_g_snpg_samqc.RData")

```

Here is the list of samples that were removed:

```{r removing table, echo=FALSE}

removed.ids <- bad.sample[c(2)]
names(removed.ids) <- "Neogen_Sample_ID"

rownames(removed.ids) <- NULL

removed.ids %>%
  kable(escape = F,align = c("c"),linesep ="\\hline") %>%
  kable_styling(full_width = F) %>%
  kable_styling("striped", full_width = F) 

```

Below is a table summarising the problematic samples found throughout QC. These include the following:

+ no_pheno == sample does not have a phenotype
+ high_miss == sample has higher than 10% missing genotypes 
+ diff_sex == sample has sex mismatch
+ high_xo == sample had crossover events greater than 1000 and less than 200
+ high_geno_errors == Sample has geno errors above 1%
+ DO_generation == DO generation is 1
+ dublicate_ID == sample was duplicated

**NB:** For duplcate pairs, the one that was chosen to be removed was the one that had a higher missing rate

```{r removing samples table, echo=FALSE}


gm.pheno = data.frame(id=rownames(gm$pheno),gm$pheno)
qc_info <- merge(qc_info,gm.pheno,by = "id")

#bad.sample <- qc_info[qc_info$ngen ==1 | qc_info$Number_crossovers <= 200 | qc_info$Number_crossovers >=1000 | qc_info$percent_missing >= 10 | qc_info$genotype_erros >= 1 | qc_info$remove.id.duplicated == TRUE | is.na(qc_info$Time_min)==TRUE | is.na(qc_info$Distance_m)==TRUE | is.na(qc_info$Energy_J)==TRUE,]

ind1<-summary.cg[c("ind1","iid1")]
names(ind1) <- c("id","iid")
ind2<-summary.cg[c("ind2","iid2")]
names(ind2) <- c("id","iid")
dup.IDS <- rbind(ind1, ind2)
dup.IDS <-dup.IDS[!duplicated(dup.IDS),]
dup.IDS$duplicated <- "TRUE"

bad.sample <- merge(qc_info,dup.IDS,by="id",all=T)


bad <- NULL
bad$Thaiss_ID <- bad.sample$Thaiss.ID
bad$Neogen_Sample_ID <- bad.sample$Tube.ID
bad$no_pheno <- ifelse(is.na(bad.sample$Time_min)==TRUE | is.na(bad.sample$Distance_m)==TRUE | is.na(bad.sample$Energy_J)==TRUE, 'XX', '')
bad$high_miss <- ifelse(bad.sample$percent_missing >= 10, 'XX', '')
bad$diff_sex <- ifelse(bad.sample$sex.match ==FALSE | is.na(bad.sample$sex.match) == TRUE, 'XX', '')
bad$high_xo <- ifelse(bad.sample$Number_crossovers <= 200 | bad.sample$Number_crossovers >=1000, 'XX', '')
bad$high_geno.errors <- ifelse(bad.sample$ngen ==1, 'XX', '')
bad$DO_generation <- ifelse(bad.sample$genotype_erros >= 1, 'XX', '')
bad$duplicate_ID <- ifelse(bad.sample$duplicated == TRUE, 'XX', '')

bad <- data.frame(bad)
badind <- subset(bad, 
				 bad$no_pheno == 'XX'|
				 bad$high_miss == 'XX'|
				 bad$diff_sex == 'XX'|	
				 bad$high_xo == 'XX'|
				 bad$high_geno.errors == 'XX'|
				 bad$DO_generation == 'XX'|
				 bad$duplicate_ID == 'XX')

badind[] <- lapply(badind, as.character)
badind$Thaiss_ID <- ifelse(badind$Thaiss == 994 | badind$Thaiss == 995 | badind$Thaiss == 996 |badind$Thaiss == 997 | badind$Thaiss == 998 | badind$Thaiss == 999, "--", bad$Thaiss_ID)
badind[is.na(badind)] <- ""

badind[] %>% 
   mutate(
     no_pheno = ifelse(no_pheno == 'XX',
                  cell_spec(no_pheno, color = 'green',background = 'green'),
                  ''),
     high_miss = ifelse(high_miss == 'XX',
                  cell_spec(high_miss, color = 'red',background = 'red'),
                  ''),
     diff_sex = ifelse(diff_sex == 'XX',
                  cell_spec(diff_sex, color = 'blue',background = 'blue'),
                  ''),
     high_xo = ifelse(high_xo == 'XX',
                  cell_spec(high_xo, color = 'purple',background = 'purple'),
                  ''),
     high_geno.errors = ifelse(high_geno.errors == 'XX',
                  cell_spec(high_geno.errors, color = 'purple',background = 'purple'),
                  ''),
     DO_generation = ifelse(DO_generation == 'XX',
                  cell_spec(DO_generation, color = 'light blue',background = 'ligh blue'),
                  ''),
     duplicate_ID = ifelse(duplicate_ID == 'XX',
                  cell_spec(duplicate_ID, color = 'orange',background = 'orange'),
                  '')
     ) %>%
   kable(escape = F,align = c("ccccccccc"),linesep ="\\hline") %>%
   kable_styling("striped", full_width = F) %>%
   column_spec(1:9, width = "3cm") 

```

After full qc (sample & [SNP](2.3_snp_qc.html)), we ran all checks [again](2.4_afterqc.html) to see if data improved for `R/qtl2` analysis. 
