---
title: "Step 2: SNP QC"
author: "Belinda Cornes"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

#setwd("/projects/compsci/corneb_cs/workflowr/Thaiss/Thaiss_workflowr_2/")

#libraries
library(broman)
library(qtl2)
library(qtlcharts)
library(ggplot2)
library(ggrepel)
#library(DOQTL)
library(mclust)
#source("code/reconst_utils.R")
library("kableExtra")
library("knitr")
library("fst")
```

### Loading Project
```{r loading project}
load("data/e_g_snpg_samqc.RData")
gm <- get(load("data/gm_DO112_samqc.RData"))

gm
```

It can also be useful to look at the proportion of missing genotypes by marker. Markers with a lot of missing data were likely difficult to call, and so the genotypes that were called may contain a lot of errors.

### Marker Missing Data 
```{r Missing data in Markers and Genotype frequencies Markers, fig.height = 6, fig.width = 9.5, fig.align = "center"}

pmis_mar <- n_missing(gm, "marker", "proportion")*100
save(pmis_mar, file = "data/percent_missing_marker.RData")

par(mar=c(5.1,0.6,0.6, 0.6))
hist(pmis_mar, breaks=seq(0, 100, length=201),
     main="", yaxt="n", ylab="", xlab="Percent missing genotypes")
rug(pmis_mar)

pdf(file = "output/Percent_missing_genotype_data_per_marker.pdf")
par(mar=c(5.1,0.6,0.6, 0.6))
hist(pmis_mar, breaks=seq(0, 100, length=201),
     main="", yaxt="n", ylab="", xlab="Percent missing genotypes")
rug(pmis_mar)
dev.off()
```

```{r missing table, echo=FALSE}

pmis <- NULL
pmis$pmis_mar_5 <- sum(pmis_mar >= 5)
pmis$pmis_mar_10 <- sum(pmis_mar >= 10)
pmis$pmis_mar_15 <- sum(pmis_mar >= 15)
pmis$pmis_mar_25 <- sum(pmis_mar >= 25)
pmis$pmis_mar_50 <- sum(pmis_mar >= 50)
pmis$pmis_mar_75 <- sum(pmis_mar >= 75)
pmis$total_snps <- nrow(as.data.frame(pmis_mar))

pmis <- t(as.data.frame(pmis))
pmis <- as.data.frame(pmis)
pmis$count <- pmis$V1

pmis[c(2)] %>%
  kable(escape = F,align = c("ccccccccc"),linesep ="\\hline") %>%
  kable_styling(full_width = F) %>%
  kable_styling("striped", full_width = F)  %>%
  row_spec(7 ,bold=T,color= "white",background = "black")

```

For the markers with lots of missing genotypes, itâ€™s not necessarily the case that the remaining genotypes are full of errors, but in studying the allele intensities at these SNPs, it does appear that for the bulk of such markers, the genotypes are not being called appropriately.

### Marker Genotype Frequencies
```{r Genotype frequencies Markers, fig.height = 6, fig.width = 9.5, fig.align = "center"}

g <- do.call("cbind", gm$geno[1:19])
fg <- do.call("cbind", gm$founder_geno[1:19])
g <- g[,colSums(fg==0)==0]
fg <- fg[,colSums(fg==0)==0]
fgn <- colSums(fg==3)

gf_mar <- t(apply(g, 2, function(a) table(factor(a, 1:3))/sum(a != 0)))
gn_mar <- t(apply(g, 2, function(a) table(factor(a, 1:3))))

pdf(file = "output/genotype_frequency_marker.pdf")
par(mfrow=c(2,2), mar=c(0.6, 0.6, 2.6, 0.6))
for(i in 1:4) {
  triplot(c("AA", "AB", "BB"), main=paste0("MAF = ", i, "/8"))
  z <- gf_mar[fgn==i,]
  z <- z[rowSums(is.na(z)) < 3,]
  tripoints(z, pch=21, bg="gray80", cex=0.6)
  tripoints(c((1-i/8)^2, 2*i/8*(1-i/8), (i/8)^2), pch=21, bg="violetred")
}
dev.off()

par(mfrow=c(2,2), mar=c(0.6, 0.6, 2.6, 0.6))
for(i in 1:4) {
  triplot(c("AA", "AB", "BB"), main=paste0("MAF = ", i, "/8"))
  z <- gf_mar[fgn==i,]
  z <- z[rowSums(is.na(z)) < 3,]
  tripoints(z, pch=21, bg="gray80", cex=0.6)
  tripoints(c((1-i/8)^2, 2*i/8*(1-i/8), (i/8)^2), pch=21, bg="violetred")
}

save(gf_mar, file = "data/genotype_freq_marker.RData")

```

The bulk of the markers seem well behaved, but there are a number of markers with unusual genotype frequencies. There are markers that show no homozygotes for the major allele. (These sit on the left edge.) There are markers that are monomorphic (100% AA genotypes; lower-right vertex). And there some markers with some of each homozygote but no heterozygotes (bottom edge). Further, there are a bunch of markers where the minor allele appears to be private to one founder strain (upper-left panel) but show a high frequency of minor alleles in the DO offspring. 


### Marker Genotype Errors
```{r Genotype errors Markers, fig.height = 6, fig.width = 9.5, fig.align = "center"}
errors_mar <- colSums(e>2)/n_typed(gm, "marker")*100

grayplot(pmis_mar, errors_mar,
         xlab="Proportion missing", ylab="Proportion genotyping errors")

pdf(file = "output/genotype_error_marker.pdf")
grayplot(pmis_mar, errors_mar,
         xlab="Proportion missing", ylab="Proportion genotyping errors")
dev.off()

save(errors_mar, file = "data/genotype_errors_marker.RData")

```

Markers with higher rates of missing genotypes tend to show higher errors rates.

### Removing Markers
```{r listing markers, echo=FALSE}

#Missingness

length(pmis_mar[pmis_mar >= 10])

high_miss <- find_markerpos(gm, names(pmis_mar[pmis_mar >= 10]))
high_miss$id <- rownames(high_miss)
high_miss_df <- as.data.frame(pmis_mar[pmis_mar >= 10])
high_miss_df$index = 1: nrow(high_miss_df)
high_miss_df$id <- rownames(high_miss_df)

high_miss_bad <- merge(high_miss,high_miss_df, by=c("id"),all=T)
names(high_miss_bad)[5] <- c("high_miss")
names(high_miss_bad)[1] <- c("marker")
high_miss_bad <- high_miss_bad[order(high_miss_bad$index),]


#Monomorphic/Low Frequency markers

low_freq_df <- as.data.frame(gf_mar)
low_freq_df$count <- rowSums(low_freq_df <= 0.01)
low_freq_df <- low_freq_df[low_freq_df$count == 2,]
low_freq_df$id <- rownames(low_freq_df)
low_freq_df$index = 1: nrow(low_freq_df)
low_freq <- find_markerpos(gm, rownames(low_freq_df))
low_freq$id <- rownames(low_freq)

nrow(low_freq)

low_freq_bad <- merge(low_freq,low_freq_df, by=c("id"),all=T)
names(low_freq_bad)[5] <- c("AA_freq")
names(low_freq_bad)[6] <- c("AB_freq")
names(low_freq_bad)[7] <- c("BB_freq")
names(low_freq_bad)[1] <- c("marker")
low_freq_bad <- low_freq_bad[order(low_freq_bad$index),]

##Genotyping Error

length(errors_mar[errors_mar > 5])

error_markers <- find_markerpos(gm, names(errors_mar[errors_mar > 5]))
error_markers$id <- rownames(error_markers)
#rne <- rownames(as.data.frame(errors_mar))
error_mars_df <- as.data.frame(errors_mar[errors_mar > 5])
error_mars_df$id = rownames(error_mars_df)
error_mars_df$index = 1: nrow(error_mars_df)

error_markers_bad <- merge(error_markers,error_mars_df, by=c("id"),all=T)
names(error_markers_bad)[5] <- c("error_mars")
names(error_markers_bad)[1] <- c("marker")
error_markers_bad <- error_markers_bad[order(error_markers_bad$index),]


### merge all

bad_markers <- rbind(high_miss_bad[c("marker","chr","gmap","pmap")], low_freq_bad[c("marker","chr","gmap","pmap")], error_markers_bad[c("marker","chr","gmap","pmap")])
nrow(bad_markers)

duplicate <- bad_markers[duplicated(bad_markers),]

bad_markers <- bad_markers[!duplicated(bad_markers),]

save(bad_markers, file = "data/bad_markers_all.RData")

```
List of markers with errors can be found [here](bad_markers_allqc.html)

### Array Intensities

Figures for all SNPs that are will be removed are found [here](array_intensities_bad.markers_allqc.html)


```{r Removing markers }

#missing in at least 10% of the samples


gm_DO112_allqc2 <- drop_markers(gm_DO112_samqc, bad_markers$marker)
gm_DO112_allqc <- drop_nullmarkers(gm_DO112_allqc2)


gm_DO112_allqc

save(gm_DO112_allqc, file = "data/gm_DO112_allqc.RData")

```

After full qc ([sample](2.2_sample_bqc.html) $ SNP), we ran all checks [again](2.4.1_afterqc.html) to see if data improved for `R/qtl2` analysis.
