---
title: "Results from `R/qtl2` Analysis (Plots)"
author: "Belinda Cornes"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This script will plot the qtl mapping results.

```{r library, message=FALSE, echo=FALSE}

library(qtl2)
library(ggplot2)

#setwd("/Users/corneb/Documents/MyJax/CS/Projects/qtl2/Univ_of_Penn_Thaiss/Thaiss/three.batches/Thaiss_workflowr_old")

gm_DO112_allqc <- get(load("data/gm_DO112_allqc.RData"))

gm_DO112_allqc

```

### Plot Scan Results: m1 (Sex+Generation)
```{r plot scan result for m1 Univ_of_Penn_Thaiss_pheno_all_allqc, fig.height=14, fig.width=14, eval=TRUE}
pheno <- read.csv("data/Univ_of_Penn_Thaiss_pheno_all.csv", header = TRUE)

pheno[1:10,]

print("m1")
load("output/m1.Univ_of_Penn_Thaiss_pheno_all_allqc.RData")

#permutation results
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.
# operm <- list()
# for(i in 1:100){
#   operm[[i]] <- get(load(paste0("output/permu_rnorm/m2_Univ_of_Penn_Thaiss_pheno_all_allqc_", colnames(pheno)[2], "_", i, ".RData")))
#   #operm[[i]] <- operm[[i]]$OFA.total.distance.traveled
# }
# operm <- matrix(do.call(rbind, operm[3:5]),ncol = 1)
# class(operm) <- c("scan1perm", "matrix")
# cutoff <- summary(operm, c(0.05, 0.1))[,1]

#genome-wide results
pdf(file = paste0("output/m1.Univ_of_Penn_Thaiss_pheno_all_allqc", "_genomescan.pdf"), width = 14)
for(i in names(m1.qtl.out)){
  par(mar=c(5.1, 4.1, 1.1, 1.1))
  ymx <- maxlod(m1.qtl.out[[i]]) # overall maximum LOD score
  plot(m1.qtl.out[[i]], gm_DO112_allqc$gmap, lodcolumn=1,
       col="slateblue", ylim=c(0, ymx*1.02),
       main=i)
#   abline(h=cutoff[1], col="red",lty=2) # 0.05 cutoff
#   abline(h=cutoff[2], col="red",lty=1) # 0.1 cutoff
}
dev.off()

for(i in names(m1.qtl.out)){
  par(mar=c(5.1, 4.1, 1.1, 1.1))
  ymx <- maxlod(m1.qtl.out[[i]]) # overall maximum LOD score
  plot(m1.qtl.out[[i]], gm_DO112_allqc$gmap, lodcolumn=1,
       col="slateblue", ylim=c(0, ymx*1.02),
       main=i)
#   abline(h=cutoff[1], col="red",lty=2) # 0.05 cutoff
#   abline(h=cutoff[2], col="red",lty=1) # 0.1 cutoff
}

#load blup results
x <- list()
for(i in colnames(pheno)[2:5]){
  y <- get(load(paste0("output/blup_allqc/m1_blup_Univ_of_Penn_Thaiss_pheno_all_",i,"_allqc.RData")))
  x[[i]] <- y[[i]]
}
m1.blup = x

#coeffects plot
#pdf(file = paste0("output/m1.Univ_of_Penn_Thaiss_pheno_all_allqc", "_coeffgeneplot.pdf"), width = 14)
for(i in names(m1.qtl.out)){
  chr <- as.character(m1.peak[[i]][m1.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m1.coef[[i]], 
              gm_DO112_allqc$gmap[chr], 
              scan1_output=m1.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =i)
  plot_coefCC(m1.blup[[i]], 
              gm_DO112_allqc$gmap[chr], 
              scan1_output=m1.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =i)
  plot(m1.snps[[i]]$lod, 
       m1.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m1.genes[[i]],
       main = i)
}
#dev.off()
```

### Plot Scan Results: m2 (Sex)
```{r plot scan result for m2 Univ_of_Penn_Thaiss_pheno_all_allqc,fig.height=14, fig.width=14, eval=TRUE}

print("m2")
load("output/m2.Univ_of_Penn_Thaiss_pheno_all_allqc.RData")

#permutation results
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.
# operm <- list()
# for(i in 3:5){
#   operm[[i]] <- get(load(paste0("output/permu_rnorm/m2_Univ_of_Penn_Thaiss_pheno_all_allqc_", colnames(pheno)[14], "_", i, ".RData")))
#   operm[[i]] <- operm[[i]]$OFA.total.distance.traveled
# }
# operm <- matrix(do.call(rbind, operm[3:5]),ncol = 1)
# class(operm) <- c("scan1perm", "matrix")
# cutoff <- summary(operm, c(0.05, 0.1))[,1]

#genome-wide results
pdf(file = paste0("output/m2.Univ_of_Penn_Thaiss_pheno_all_allqc", "_genomescan.pdf"), width = 14)
for(i in names(m2.qtl.out)){
  par(mar=c(5.1, 4.1, 1.1, 1.1))
  ymx <- maxlod(m2.qtl.out[[i]]) # overall maximum LOD score
  plot(m2.qtl.out[[i]], gm_DO112_allqc$gmap, lodcolumn=1,
       col="slateblue", ylim=c(0, ymx*1.02),
       main=i)
  # abline(h=cutoff[1], col="red",lty=2) # 0.05 cutoff
  # abline(h=cutoff[2], col="red",lty=1) # 0.1 cutoff
}
dev.off()

for(i in names(m2.qtl.out)){
  par(mar=c(5.1, 4.1, 1.1, 1.1))
  ymx <- maxlod(m2.qtl.out[[i]]) # overall maximum LOD score
  plot(m2.qtl.out[[i]], gm_DO112_allqc$gmap, lodcolumn=1,
       col="slateblue", ylim=c(0, ymx*1.02),
       main=i)
  # abline(h=cutoff[1], col="red",lty=2) # 0.05 cutoff
  # abline(h=cutoff[2], col="red",lty=1) # 0.1 cutoff
}

#load blup results
x <- list()
for(i in colnames(pheno)[2:5]){
  y <- get(load(paste0("output/blup_allqc/m2_blup_Univ_of_Penn_Thaiss_pheno_all_",i,"_allqc.RData")))
  x[[i]] <- y[[i]]
}
m2.blup = x

#coeffects plot
#pdf(file = paste0("output/m2.Univ_of_Penn_Thaiss_pheno_all_allqc", "_coeffgeneplot.pdf"), width = 14)
for(i in names(m2.qtl.out)){
  chr <- as.character(m2.peak[[i]][m2.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m2.coef[[i]], 
              gm_DO112_allqc$gmap[chr], 
              scan1_output=m2.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =i)
  plot_coefCC(m2.blup[[i]], 
              gm_DO112_allqc$gmap[chr], 
              scan1_output=m2.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =i)
  plot(m2.snps[[i]]$lod, 
       m2.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m2.genes[[i]],
       main = i)
}
#dev.off()
```

### Plot Scan Results: m3 (Sex and intcovar)
```{r plot scan result for m3 Univ_of_Penn_Thaiss_pheno_all_allqc, fig.height=14, fig.width=14, eval=TRUE}
print("m3")
load("output/m3.Univ_of_Penn_Thaiss_pheno_all_allqc.RData")

#permutation results
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.
# operm <- list()
# for(i in 3:5){
#   operm[[i]] <- get(load(paste0("output/permu_rnorm/m3_Univ_of_Penn_Thaiss_pheno_all_allqc_", colnames(pheno)[14], "_", i, ".RData")))
#   operm[[i]] <- operm[[i]]$OFA.total.distance.traveled
# }
# operm <- matrix(do.call(rbind, operm[3:5]),ncol = 1)
# class(operm) <- c("scan1perm", "matrix")
# cutoff <- summary(operm, c(0.05, 0.1))[,1]

#genome-wide results
pdf(file = paste0("output/m3.Univ_of_Penn_Thaiss_pheno_all_allqc", "_genomescan.pdf"), width = 14)
for(i in names(m3.qtl.out)){
  par(mar=c(5.1, 4.1, 1.1, 1.1))
  ymx <- maxlod(m3.qtl.out[[i]]) # overall maximum LOD score
  plot(m3.qtl.out[[i]], gm_DO112_allqc$gmap, lodcolumn=1,
       col="slateblue", ylim=c(0, ymx*1.02),
       main=i)
  # abline(h=cutoff[1], col="red",lty=2) # 0.05 cutoff
  # abline(h=cutoff[2], col="red",lty=1) # 0.1 cutoff
}
dev.off()

for(i in names(m3.qtl.out)){
  par(mar=c(5.1, 4.1, 1.1, 1.1))
  ymx <- maxlod(m3.qtl.out[[i]]) # overall maximum LOD score
  plot(m3.qtl.out[[i]], gm_DO112_allqc$gmap, lodcolumn=1,
       col="slateblue", ylim=c(0, ymx*1.02),
       main=i)
  # abline(h=cutoff[1], col="red",lty=2) # 0.05 cutoff
  # abline(h=cutoff[2], col="red",lty=1) # 0.1 cutoff
}

#load blup results
x <- list()
for(i in colnames(pheno)[2:5]){
  y <- get(load(paste0("output/blup_allqc/m3_blup_Univ_of_Penn_Thaiss_pheno_all_",i,"_allqc.RData")))
  x[[i]] <- y[[i]]
}
m3.blup = x

#coeffects plot
#pdf(file = paste0("output/m3.Univ_of_Penn_Thaiss_pheno_all_allqc", "_coeffgeneplot.pdf"), width = 14)
for(i in names(m3.qtl.out)){
  chr <- as.character(m3.peak[[i]][m3.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m3.coef[[i]], 
              gm_DO112_allqc$gmap[chr], 
              scan1_output=m3.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =i)
  plot_coefCC(m3.blup[[i]], 
              gm_DO112_allqc$gmap[chr], 
              scan1_output=m3.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =i)
  plot(m3.snps[[i]]$lod, 
       m3.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m3.genes[[i]],
       main = i)
}
#dev.off()
```
